mass_mailing_resend/models/mass_mailing.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# Copyright 2016 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import _, exceptions, models
+
+
+class MailMassMailingList(models.Model):
+    _inherit = "mail.mass_mailing"
+
+    def button_draft(self):
+        """Return to draft state for resending the mass mailing."""
+        if any(self.mapped(lambda x: x.state != 'done')):
```

Comment:
`if self.filtered(...):`, or use a normal `for` loop, which is more performant since it would stop at 1st match.

Reply:
In this case, as the list is going to be very short, I prefer this formula to avoid one cyclomatic level more.

Reply:
In such case, use `if self.filtered(...)`